test_collections/matter/sdk_tests/support/chip/chip_server.py

```diff
@@ -158,19 +158,6 @@ async def start(
 
         return cast(Generator, self.__server_logs)
 
-    def __wait_for_server_exit(self) -> Optional[int]:
-        if self.__chip_server_id is None:
-            self.logger.info(
-                "Server execution id not found, cannot wait for server exit."
-            )
-            return None
-
-        exit_code = self.sdk_container.exec_exit_code(self.__chip_server_id)
-        while exit_code is None:
-            exit_code = self.sdk_container.exec_exit_code(self.__chip_server_id)
```

Comment:
Instead of removing the wait completely... wouldn't it be interesting to add a timeout here?

That way, we would at least wait for the timeout. Since this problem is specific to a test case.

I believe it will have fewer side effects :)